packages/core/docs/guide/checkout.md

```diff
@@ -0,0 +1,324 @@
+
+### Extending `VsfShippingProvider` and reacting to it's events
+We can pass asynchronous functions to the `VsfShippingProvider` component to hook into different events within its lifecycle, override initial function parameters or react to specific events like method selection. We will call these methods "hooks".
+
+Because every shipping provider is different, not all of them are present in every integration. Always refer to the documentation of a specific provider to learn which hooks are available.
+
+- **beforeLoad** `(config => config)` - Called before loading shipping methods.
+- **afterLoad** `(shippingMethodsResponse => shippingMethodsResponse.shippingMethods)` - Called after loading shipping methods.
+- **beforeSelect** `(shippingMethod => shippingMethod)` - Called before selecting shipping method.
+- **afterSelect** `(selectedShippingMethod => void)` - Called after selecting shipping method.
+- **beforeSelectedDetailsChange** `(details => details)` - Called before modifying currently picked shipping method, e.g. selecting parcel locker on the map.
+- **afterSelectedDetailsChange** `(details => void)` - Called after modifying currently picked shipping method.
+- **onError** `(({ action, error }) => void)` - Called when some operation throws an error.
+
+```vue
+<VsfShippingProvider
+  :beforeLoad="beforeLoad"
+  :afterLoad="afterLoad"
+  :beforeSelect="beforeSelect"
+  :afterSelect="afterSelect"
+  :beforeSelectedDetailsChange="beforeSelectedDetailsChange"
+  :afterSelectedDetailsChange="afterSelectedDetailsChange"
+  :onError="onError"
+/>
+```
+## Collecting and saving billing details
+Billing details are information about the payer's address used by store owners to prepare invoices and payment providers to evaluate the probability of fraud payment.
+
+We can load billing details by calling the `load` method in `useBilling` composable and use the `billing` property to access it.
+```js{8,16}
+import { useBilling } from '{INTEGRATION}';
+import { onSSR } from '@vue-storefront/core';
+
+export default {
+  setup () {
+    const {
+      load,
+      billing
+    } = useBilling();
+
+    onSSR(async () => {
+      await load();
+    });
+
+    return {
+      billing
+    };
+  }
+}
+```
+`billing` property returns `null` if the `load` function was not invoked or nothing is saved.   
+
+We can use `save` method to save billing details.
+```vue{2,15,24}
+<template>
+  <form @submit.prevent="save({ billingDetails: billingForm })">
+    <!-- form fields -->
+    <button type="submit" :disabled="loading">{{ $t('Submit') }}</button>
+  </form>
+</template>
+
+<script>
+import { useBilling } from '{INTEGRATION}';
+import { onSSR } from '@vue-storefront/core';
+
+export default {
+  setup () {
+    const {
+      save,
+      loading,
+      billing
+    } = useBilling();
+
+    const billingForm = ref(/* object for billing details */);
+
+    return {
+      billingForm,
+      save
+    };
+  }
+}
+</script>
+```
+
+## Making an order
```

Comment:
I know in many cases the order is opposite but maybe we should have this section at the end?

Reply:
I am not sure about that. In PaymentProvider part, I mention about doing payment after order and using `useMakeOrder` which requires to put Order part before.